.travis.yml

```diff
@@ -7,37 +7,44 @@ branches:
   - trying.tmp
 
 env:
-  # Should pandas tests be removed or replaced wih import checks?
-  #- UNCERTAINTIES="N" PYTHON="3.6" NUMPY_VERSION=1.14 PANDAS=1
-  - UNCERTAINTIES="N" PYTHON="3.3" NUMPY_VERSION=1.9.2 PANDAS=0
-  - UNCERTAINTIES="N" PYTHON="3.4" NUMPY_VERSION=1.11.2 PANDAS=0
-  - UNCERTAINTIES="N" PYTHON="3.5" NUMPY_VERSION=1.11.2 PANDAS=0
-  - UNCERTAINTIES="Y" PYTHON="3.5" NUMPY_VERSION=1.11.2 PANDAS=0
-  - UNCERTAINTIES="N" PYTHON="3.6" NUMPY_VERSION=1.11.2 PANDAS=0
-  - UNCERTAINTIES="N" PYTHON="2.7" NUMPY_VERSION=0 PANDAS=0
-  - UNCERTAINTIES="N" PYTHON="3.5" NUMPY_VERSION=0 PANDAS=0
-  # Test with the latest numpy version
-  - UNCERTAINTIES="N" PYTHON="2.7" NUMPY_VERSION=1.14 PANDAS=0
-  #- UNCERTAINTIES="N" PYTHON="3.4" NUMPY_VERSION=1.14 PANDAS=0
-  - UNCERTAINTIES="N" PYTHON="3.5" NUMPY_VERSION=1.14 PANDAS=0
-  - UNCERTAINTIES="Y" PYTHON="3.5" NUMPY_VERSION=1.14 PANDAS=0
-  - UNCERTAINTIES="N" PYTHON="3.6" NUMPY_VERSION=1.14 PANDAS=0
+  # This project adheres to NEP-29
+  # https://numpy.org/neps/nep-0029-deprecation_policy.html
+
+  # Refer to https://docs.scipy.org/doc/numpy/release.html for
+  # min/max Python version supported by numpy
+  # Refer to history of https://github.com/lebigot/uncertainties/blob/master/setup.py
+  # for min/max Python versions supported by uncertainties
+  # At the moment of writing, uncertainties doesn't officially support Python 3.8
+
+  # Without numpy, without uncertainties
+  - UNCERTAINTIES=0   PYTHON=3.6 NUMPY_VERSION=0 PANDAS=0
+  - UNCERTAINTIES=0   PYTHON=3.7 NUMPY_VERSION=0 PANDAS=0
+  - UNCERTAINTIES=0   PYTHON=3.8 NUMPY_VERSION=0 PANDAS=0
+  # Without numpy, min version of uncertainties
+  - UNCERTAINTIES=3.0 PYTHON=3.6 NUMPY_VERSION=0 PANDAS=0
+  - UNCERTAINTIES=3.0 PYTHON=3.7 NUMPY_VERSION=0 PANDAS=0
+  # Min version of numpy, without uncertainties
+  - UNCERTAINTIES=0   PYTHON=3.6 NUMPY_VERSION=1.14 PANDAS=0
+  - UNCERTAINTIES=0   PYTHON=3.7 NUMPY_VERSION=1.14 PANDAS=0
+  - UNCERTAINTIES=0   PYTHON=3.8 NUMPY_VERSION=1.17 PANDAS=0
+  # Min version of numpy, min version of uncertainties
+  - UNCERTAINTIES=3.0 PYTHON=3.6 NUMPY_VERSION=1.14 PANDAS=0
+  - UNCERTAINTIES=3.0 PYTHON=3.7 NUMPY_VERSION=1.14 PANDAS=0
+  # Latest versions of numpy and uncertainties (where different from the minimum ones)
+  - UNCERTAINTIES=3.1 PYTHON=3.6 NUMPY_VERSION=1.17 PANDAS=0
+  - UNCERTAINTIES=3.1 PYTHON=3.7 NUMPY_VERSION=1.17 PANDAS=0
```

Comment:
This envs combination may be overkill. For example, Testing Python 3.7 in the combinations where there are already 3.6 and 3.8 is probably overkill - please share your opinions.